src/extensions/contact-loaders/civicrm/util.js

```diff
@@ -0,0 +1,328 @@
+      is_deceased: { "=": 0 },
+      is_opt_out: { "=": 0 },
+      contact_type: "Individual",
+
+      return: [
+        "id",
+        "first_name",
+        "last_name",
+        "postal_code",
+        ...customFieldNames
+      ],
+
+      "api.Phone.get": {
+        contact_id: "$value.id",
+        phone_type_id: "Mobile",
+
+        return: ["id", "phone_numeric"],
+        options: { limit: 1 }
+      },
+      // Closest thing to docs for this: https://lab.civicrm.org/dev/core/blob/d434a5cfb2dc3c248ac3c0d8570bd8e9d828f6ad/api/v3/Contact.php#L403
+      group: groupId
+    },
+    callback
+  );
+  return paginatedData;
+}
+
+export async function addContactToGroup(contactId, groupId) {
+  const config = getCivi();
+
+  const res = await fetchfromAPI(
+    config,
+    "GroupContact",
+    {
+      contact_id: contactId,
+      group_id: groupId
+    },
+    "create",
+    { method: "post" }
+  );
+
+  return res;
+}
+
+export async function sendEmailToContact(contactId, templateId) {
+  const config = getCivi();
+
+  const res = await fetchfromAPI(
+    config,
+    "Email",
+    {
+      contact_id: contactId,
+      template_id: templateId
+    },
+    "send",
+    { method: "post" }
+  );
+
+  return res;
+}
+
+export async function addContactToTag(contactId, tagId) {
+  const config = getCivi();
+
+  const res = await fetchfromAPI(
+    config,
+    "EntityTag",
+    {
+      entity_id: contactId,
+      entity_table: "civicrm_contact",
+      tag_id: tagId
+    },
+    "create",
+    { method: "post" }
+  );
+
+  return res;
+}
+
+/**
+ * @returns {Promise<{ name: string; id: number }[]>}
+ */
+export async function searchTags() {
+  const config = getCivi();
+
+  const res = await fetchfromAPI(config, "tag", {
+    sequential: 1,
+    return: ["id", "name"],
+    options: { limit: 0 }
+  });
+  if (res) {
+    return res;
+  }
+  return [];
+}
+
+/**
+ * @returns {Promise<{ name: string; id: number }[]>}
+ */
+export async function searchEvents() {
+  const config = getCivi();
+  const currentNow = moment().format();
+  const res = await fetchfromAPI(config, "event", {
+    sequential: 1,
+    return: ["id", "title", "default_role_id", "start_date"],
+    title: { "!=": "" },
+    is_monetary: 0,
+    requires_approval: 0,
+    end_date: {
+      ">": currentNow
+    },
+    options: { limit: 0 }
+  });
+  if (res) {
+    return res;
+  }
+  return [];
+}
+
+export async function registerContactForEvent(contactId, eventId, roleId) {
+  const config = getCivi();
+
+  const res = await fetchfromAPI(
+    config,
+    "Participant",
+    {
+      contact_id: contactId,
+      event_id: eventId,
+      role_id: roleId
+    },
+    "create",
+    { method: "post" }
+  );
+
+  return res;
+}
+
+export async function searchMessageTemplates() {
+  const config = getCivi();
+  const res = await fetchfromAPI(config, "MessageTemplate", {
+    sequential: 1,
+    return: ["id", "msg_title"],
+    msg_title: { "!=": "" },
```

Comment:
What is `{ "!=": "" }`?

Reply:
It's how we exclude email message templates in CiviCRM that have a blank title from being included in the results. Blank titled templates would render as "empty" selections in the dropdown when configuring the action.

We do this to avoid ambiguity for the end user.